core-services/src/main/java/org/zalando/nakadi/service/publishing/EventOwnerExtractor.java

```diff
@@ -0,0 +1,12 @@
+package org.zalando.nakadi.service.publishing;
+
+import org.json.JSONObject;
+import org.zalando.nakadi.domain.EventOwnerHeader;
+import org.zalando.nakadi.domain.NakadiMetadata;
+
+public interface EventOwnerExtractor {
+
+    EventOwnerHeader extractEventOwner(JSONObject event);
+
+    EventOwnerHeader extractEventOwner(NakadiMetadata metadata);
```

Comment:
it would be nice to have a single method, but at the same time I realise that `EventOwnerHeader extractEventOwner(NakadiMetadata metadata);` can be later reused to work with other types of the binary.

Reply:
Potentially we can expose `metadata` part of a `JSONObject` via some common interface...  Then a single method will be possible.